modules/platforms/dotnet/Apache.Ignite/Internal/Table/Serialization/ObjectSerializerHandler.cs

```diff
@@ -500,11 +500,13 @@ private static void EmitFieldRead(FieldInfo? fieldInfo, ILGenerator il, Column c
         private static void ValidateFieldType(FieldInfo fieldInfo, Column column)
         {
             var columnType = column.Type.ToType();
+
             var fieldType = Nullable.GetUnderlyingType(fieldInfo.FieldType) ?? fieldInfo.FieldType;
+            fieldType = fieldType.IsEnum ? Enum.GetUnderlyingType(fieldType) : fieldType;
 
             if (fieldType != columnType)
             {
-                var message = $"Can't map field '{fieldInfo.DeclaringType?.Name}.{fieldInfo.Name}' of type '{fieldType}' " +
+                var message = $"Can't map field '{fieldInfo.DeclaringType?.Name}.{fieldInfo.Name}' of type '{fieldInfo.FieldType}' " +
```

Comment:
Just an idea - provide a list of supported types or a link to a future documentation link?

Reply:
In this case the issue is not about unsupported type, but about type mismatch.

Unlike Ignite 2.x, you can't put an arbitrary object into the table (cache). First, you create a table with SQL. Then you can map a class to that table. And if you make a mistake with field types, you'll get a message like "Can't map int field 'foo' to string column 'foo'". So you go ahead and change the field type. No need to check the docs.